code/modules/admin/verbs/serialization.dm

```diff
@@ -11,7 +11,73 @@
 		return
 
 	var/atom/movable/AM = holder.marked_datum
-	to_chat(src, json_encode(AM.serialize()))
+
+	var/json_data = json_encode(AM.serialize())
+
+	var/choice = alert(usr, "Would you like to store this on your PC or server side?", "Storage Location", "PC", "Server", "Cancel")
+	if(!choice || choice == "Cancel")
+		return
+
+	if(choice == "PC")
+		to_chat(src, json_data)
+
+	if(choice == "Server")
+		// Right, get their slot names
+		var/list/slots = list("--NEW--")
+		var/datum/db_query/dbq = SSdbcore.NewQuery("SELECT slotname FROM json_datum_saves WHERE ckey=:ckey", list("ckey" = usr.ckey))
+		if(!dbq.warn_execute())
+			qdel(dbq)
+			return
+
+		while(dbq.NextRow())
+			slots += dbq.item[1]
+
+		qdel(dbq)
+
+		var/slot_choice = input(usr, "Select a slot to update, or create a new one.", "Slot Selection") as null|anything in slots
+
+		if(!slot_choice)
+			return
+
+		if(slot_choice == "--NEW--")
+			// New slot, save to DB
+			var/chosen_slot_name = input(usr, "Name your slot", "Slot name") as null|text
+			if(!chosen_slot_name || length(chosen_slot_name) == 0)
+				return
+
+			// Sanitize the name
+			var/clean_name = sanitize(copytext(chosen_slot_name, 1, 32)) // 32 chars is your max
+
+			// And save
+			var/datum/db_query/dbq2 = SSdbcore.NewQuery("INSERT INTO json_datum_saves (ckey, slotname, slotjson) VALUES(:ckey, :slotname, :slotjson)", list(
```

Comment:
this query won't be cleaned up if it passes the next if check, add in another qdel please :D